### PR TITLE
Added Reviews to Data Export

### PR DESCRIPTION
**Pull request by aaryan01**
Hey @bkimminich ! I added Reviews in the data export. Is feedback or complaints stored in the database? Couldn't find it or did I miss something?
Thanks!

### CODE REVIEW

**Comment by J12934** [comment]
Yes they are see:

- `/models/feedback.js`
- `/models/complaint.js`

File: routes/dataExport.js [hunk 1]
```diff
@@ -31,7 +31,29 @@ module.exports = function dataExport () {
           })
           userData.orders = orders
         }
-        res.status(200).send({ userData: JSON.stringify(userData, null, 2), confirmation: 'Your data export will open in a new Browser window.' })
+
+        db.reviews.find({ $where: "this.author === '" + email + "'" }).then(reviews => {
```
Comment (J12934) [inline routes/dataExport.js:35]: `$where` String concatenation should probably not be used unless it is intended for a challenge. This looks pretty injectable for people with emails like little bobby: https://www.xkcd.com/327/

**Comment by bkimminich** [comment]
Yes, they're in the "oldschool" relational DB.

File: routes/dataExport.js [hunk 2]
```diff
@@ -31,7 +31,29 @@ module.exports = function dataExport () {
...
+          const result = utils.queryResultToJson(reviews)
```
Comment (J12934) [inline routes/dataExport.js:36]: `queryResultToJson` isn't really required here as the retuned structure doesn't get used here anyway.

File: routes/dataExport.js [hunk 3]
```diff
@@ -31,7 +31,29 @@ module.exports = function dataExport () {
...
+          const data = result.data
+          if (data.length > 0) {
+            let reviews = []
+            data.map(review => {
```
Comment (J12934) [inline routes/dataExport.js:40]: No need for the tmp reviews array.
You can assign `userData.reviews` directly to the result of `reviews.map(...)`

**Comment by J12934** [comment]
Just noticed that the other part of the data export is also all written in this style. Could merge this and update both in one go.

**Comment by aaryan01** [comment]
> Just noticed that the other part of the data export is also all written in this style. Could merge this and update both in one go.

Hey @J12934 @bkimminich, thank you for your help and suggestions. You can merge this and I'll add feedbacks, complaints etc and update the code according to your suggestions.
Thanks!

**Comment by bkimminich** [comment]
Ok, I'll merge and you just do the refactorings for both in one go, @aaryan01. Thanks! :+1:

File: routes/dataExport.js (hunk 1 above)
Comment (aaryan01) [reply routes/dataExport.js:35]: Hey @J12934, I have made the other two changes but I'm not aware of an alternative for $where. Could you suggest another way?
Thanks!

Comment (J12934) [reply routes/dataExport.js:35]: Just `db.reviews.find({ author: email })`should work fine.
See the MarsDB examples: https://github.com/c58/marsdb

**Pull request merged**